flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessageDecoderDelegate.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBufAllocator;
+
+import javax.annotation.Nullable;
+
+/**
+ * Parsers for specified netty messages.
+ */
+public interface NettyMessageDecoderDelegate {
```

Comment:
Considering the naming of `NettyMessageDecoderDelegate` and `NettyMessageClientDecoder`, I think it should be reversed. The `delegate` should generally has a real working component inside it.
So  it seems more make sense for `NettyMessageDecoderDelegate` -> `NettyMessageDecoder`, `NettyMessageClientDecoder` -> `NettyMessageDecoderDelegate`.

Reply:
I'm slightly hesitate to rename the classes since we also have `NettyMessageServerDecoder`, and if we change the name of client-side decoder to `NettyMessageDecoderDelegate`, it seems to be some kind of inconsistency. 

I think we might change the DecoderDelegate to something else?

Reply:
We might not consider the `NettyMessageServerDecoder` atm, because it is still named `NettyMessageDecoder` now. If we want to refactor the server name accordingly future, we can consider the client side then.

There is a bit difference between client and server. On server side we only have one explicit decoder in flink stack now, but on client side there actually exits two decoders explicitly which should be distinguished for better understanding.